src/librawspeed/decoders/TiffBasedDecoder.h

```diff
@@ -0,0 +1,40 @@
+/*
+    RawSpeed - RAW file decoder.
+
+    Copyright (C) 2017 Axel Waggershauser
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#pragma once
+
+#include "decoders/RawDecoder.h" // for RawDecoder
+#include "io/FileMap.h"          // for FileMap
+#include "tiff/TiffIFD.h"        // for TiffRootIFDOwner
+
+namespace RawSpeed {
+
+class TiffBasedDecoder : public RawDecoder
```

Comment:
> All the decoders based on TiffBasedDecoder are tiff-based.

Exactly! Hence, the name!?! I don't see the problem here. But I'm fine with another suitable name. The name should contain the strings "Tiff" and "Decoder" and some hint at the fact that it is not a complete decoder by itself but a base class for others. Do you have a suggestion?